storage/client_test.go

```diff
@@ -367,7 +336,12 @@ func (m *multiTestContext) addStore() {
 // gossipNodeDesc adds the node descriptor to the gossip network.
 // Mostly makes sure that we don't see a warning per request.
 func gossipNodeDesc(g *gossip.Gossip, nodeID proto.NodeID) error {
-	nodeDesc := &proto.NodeDescriptor{NodeID: nodeID}
+	nodeDesc := &proto.NodeDescriptor{
+		NodeID: nodeID,
+		// Encode the node ID in the port number so that rpcSend can figure out
+		// where requests must be sent.
+		Address: util.MakeUnresolvedAddr("localhost", fmt.Sprintf("127.0.0.1:%d", nodeID)),
```

Comment:
since this is using UnresolvedAddr pretty much as a hack to pass a string through, I'd prefer @bdarnell's suggestion literally, without the `SplitHostPort` bit.